iceoryx_posh/include/iceoryx_posh/iceoryx_posh_types.inl

```diff
@@ -49,6 +50,16 @@ log::LogStream& operator<<(log::LogStream& stream, ConnectionState value) noexce
     return stream;
 }
 
+ResourcePrefix_t iceoryxResourcePrefix(uint16_t uniqueRouDiID, iox::string<1> customizer)
+{
+    static_assert(std::is_same_v<uint16_t, std::remove_const_t<decltype(uniqueRouDiID)>>);
```

Comment:
Why? It is a `uint16_t`? If there are some type requirements on the `uniqueRouDiId` shouldn't we pack it into its own type? Maybe newtype?

Reply:
It's just prevent someone change the fixed width type and then forgetting to adjust the size of the buffer for the prefix. I hadn't a better idea to make one aware of the issue if it ever happens. Ideas welcome :)

Reply:
I went with a string in `static_assert`